File: Interview.js
const mongoose = require('mongoose');

const questionResponseSchema = new mongoose.Schema({
  question: { type: String, required: true },
  answer: String,
  rating: { type: Number, min: 1, max: 10 },
  feedback: String,
  timeTaken: Number
}, { _id: false });

const interviewSchema = new mongoose.Schema({
  studentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  collegeId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'College',
    required: true
  },
  role: {
    type: String,
    required: true
  },
  category: {
    type: String,
    enum: ['tech', 'commerce', 'hr', 'custom'],
    default: 'tech'
  },
  questionsAsked: [questionResponseSchema],
  overallRating: {
    type: Number,
    min: 0,
    max: 10
  },
  strengths: [String],
  improvements: [String],
  duration: Number,
  attemptNumber: {
    type: Number,
    default: 1
  }
}, {
  timestamps: true
});

interviewSchema.index({ studentId: 1, createdAt: -1 });
interviewSchema.index({ collegeId: 1, createdAt: -1 });
interviewSchema.index({ studentId: 1, category: 1 });

module.exports = mongoose.model('Interview', interviewSchema);

File: v2/backend/app/schemas/interview.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class InterviewResponseInput(BaseModel):
    question_text: str
    answer_text: Optional[str] = None
    rating: Optional[int] = Field(default=None, ge=1, le=10)
    feedback: Optional[str] = None
    time_taken_seconds: Optional[int] = None


class InterviewSubmitRequest(BaseModel):
    role: str
    category: str = "tech"
    overall_rating: Optional[float] = None
    strengths: Optional[str] = None
    improvements: Optional[str] = None
    duration_seconds: Optional[int] = None
    responses: list[InterviewResponseInput] = []


class InterviewAttemptResponse(BaseModel):
    id: int
    student_id: int
    college_id: int
    role: str
    category: str
    overall_rating: Optional[float] = None
    strengths: Optional[str] = None
    improvements: Optional[str] = None
    duration_seconds: Optional[int] = None
    attempt_number: int
    status: str
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}
